crates/hotshot/src/traits/networking/libp2p_network.rs

```diff
@@ -330,16 +329,59 @@ impl<M: NetworkMsg, K: SignatureKey + 'static> Libp2pNetwork<M, K> {
                 is_bootstrapped: Arc::new(AtomicBool::new(false)),
                 metrics: NetworkingMetrics::new(&*metrics),
                 topic_map,
+                node_lookup_send,
+                cur_view: Arc::new(AtomicU64::new(0)),
```

Comment:
Right, we should get a new proposal and update the current view number. 

If the `cur_view` number is somehow behind, the worst case scenario is we perform a lookup to a peer we don't need (which will eventually be cached). We can probably add checks/initialize to a more sane number, but since we require the network to be initialized to get any injected consensus info, it would be kind of a chicken and egg situation

Reply:
Awesome, this totally makes sense! It might be good to add in a comment with this argument for correctness.